midterm-2023-11-07/code/5130-stefan-vitanov/script.ts

```diff
@@ -0,0 +1,349 @@
+    return `${trimmedString}...`;
+  }
+ 
+ 
+function formatGenres(genres: string[]): string {
+    return genres.join(' / ');
+  }
+ 
+ 
+  function formatCast(cast: Cast[]): string {
+    return cast.sort((a, b) => a.actor.localeCompare(b.actor))
+               .slice(0, 5)
+               .map((member, index, arr) => {
+                 const actorName = `<strong>${member.actor}</strong>`;
+                 const characterName = ` as ${member.character}`;
+                 return index === arr.length - 1 && arr.length < 6 ? actorName + characterName : actorName + characterName + (index === arr.length - 2 ? ' &' : ', ');
+               })
+               .join('') + (cast.length > 5 ? '...' : '');
+  }
+ 
+ 
+  function formatOscars(oscars: Oscars): string {
+    const oscarEntries = Object.entries(oscars)
+      .map(([key, value]) => `${key.replace(/([A-Z])/g, ' $1').trim()} won by ${value}`);
+    return oscarEntries.sort().join('<br>');
+  }
+ 
+ 
+  function createMovieHTML(movie: Movie): HTMLElement {
+    const movieContainer = document.createElement('div');
+    movieContainer.classList.add('movie-table');
+ 
+    const idDiv = document.createElement('div');
+    idDiv.classList.add('movie-data');
+    idDiv.textContent = movie.id.toString();
+ 
+    const titleDiv = document.createElement('div');
+    titleDiv.classList.add('movie-data');
+    titleDiv.textContent = movie.title;
+ 
+    const directorDiv = document.createElement('div');
+    directorDiv.classList.add('movie-data');
+    directorDiv.textContent = movie.director;
+ 
+    const yearDiv = document.createElement('div');
+    yearDiv.classList.add('movie-data');
+    yearDiv.textContent = movie.year.toString();
+ 
+    const genreDiv = document.createElement('div');
+    genreDiv.classList.add('movie-data');
+    genreDiv.textContent = formatGenres(movie.genre);
+ 
+    const plotDiv = document.createElement('div');
+    plotDiv.classList.add('movie-data');
+    plotDiv.textContent = truncatePlot(movie.plot, 50);
+ 
+    const castDiv = document.createElement('div');
+    castDiv.classList.add('movie-data');
+    castDiv.innerHTML = formatCast(movie.cast);
+ 
+    const oscarsDiv = document.createElement('div');
+    oscarsDiv.classList.add('movie-data');
+    oscarsDiv.innerHTML = formatOscars(movie.oscars);
+ 
+ 
+    movieContainer.appendChild(idDiv);
+    movieContainer.appendChild(titleDiv);
+    movieContainer.appendChild(directorDiv);
+    movieContainer.appendChild(yearDiv);
+    movieContainer.appendChild(genreDiv);
+    movieContainer.appendChild(plotDiv);
+    movieContainer.appendChild(castDiv);
+    movieContainer.appendChild(oscarsDiv);
+ 
+    return movieContainer;
+  }
+ 
+ 
+  let currentSort: { field: keyof Movie | null, ascending: boolean } = {
+    field: null,
+    ascending: true
+  };
+ 
+  function sortMovies(movies: Movie[], sortField: keyof Movie, ascending: boolean): Movie[] {
+    return movies.sort((a, b) => {
+      let valA = a[sortField];
```

Comment:
This method is kinda too big, the complex comparisons should be extracted to their own methods. Using a method map would be even better.